homeassistant/components/light/switch.py

```diff
@@ -0,0 +1,114 @@
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Required(CONF_ENTITY_ID): cv.entity_domain(switch.DOMAIN)
+})
+
+
+async def async_setup_platform(hass: HomeAssistantType, config: ConfigType,
+                               async_add_entities,
+                               discovery_info=None) -> None:
+    """Initialize Light Switch platform."""
+    async_add_entities([LightSwitch(config.get(CONF_NAME),
+                                    config[CONF_ENTITY_ID])])
+
+
+class LightSwitch(Light):
+    """Represents a Switch as a Light."""
+
+    def __init__(self, name: str, switch_entity_id: str) -> None:
+        """Initialize Light Switch."""
+        self._name = name  # type: str
+        self._switch_entity_id = switch_entity_id  # type: str
+        self._is_on = False  # type: bool
+        self._available = False  # type: bool
+        self._async_unsub_state_changed = None
+
+    @property
+    def name(self) -> str:
+        """Return the name of the entity."""
+        return self._name
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if light switch is on."""
+        return self._is_on
+
+    @property
+    def available(self) -> bool:
+        """Return true if light switch is on."""
+        return self._available
+
+    @property
+    def should_poll(self) -> bool:
+        """No polling needed for a light switch."""
+        return False
+
+    async def async_turn_on(self, **kwargs):
+        """Forward the turn_on command to the switch in this light switch."""
+        data = {ATTR_ENTITY_ID: self._switch_entity_id}
+        await self.hass.services.async_call(
+            switch.DOMAIN, switch.SERVICE_TURN_ON, data, blocking=True)
+
+    async def async_turn_off(self, **kwargs):
+        """Forward the turn_off command to the switch in this light switch."""
+        data = {ATTR_ENTITY_ID: self._switch_entity_id}
+        await self.hass.services.async_call(
+            switch.DOMAIN, switch.SERVICE_TURN_OFF, data, blocking=True)
+
+    async def async_update(self):
+        """Query the switch in this light switch and determine the state."""
+        switch_state = self.hass.states.get(self._switch_entity_id)
+
+        if switch_state is None:
+            self._available = False
+            return
+
+        self._is_on = switch_state.state == STATE_ON
+        self._available = switch_state.state != STATE_UNAVAILABLE
+
+    async def async_added_to_hass(self) -> None:
+        """Register callbacks."""
+        @callback
+        def async_state_changed_listener(entity_id: str, old_state: State,
+                                         new_state: State):
+            """Handle child updates."""
+            self.async_schedule_update_ha_state(True)
+
+        self._async_unsub_state_changed = async_track_state_change(
+            self.hass, self._switch_entity_id, async_state_changed_listener)
+        await self.async_update()
```

Comment:
remove this and use `async_add_entities([....], True)`

Reply:
👍 Fixed.